src/errcode.h

```diff
@@ -91,24 +91,19 @@ void reset_error(struct error *err);
         }                                                       \
     } while(0)
 
-/* A variant of assert that uses our error reporting infrastructure
- * instead of aborting
+/* Assertions that use our error reporting infrastructure instead of
+ * aborting
  */
-#ifdef NDEBUG
-# define ensure(cond, obj) if (0) goto error
-# define ensure0(cond, obj) if (0) return NULL
-#else
-# define ensure(cond, obj)                                           \
+#define ensure(cond, obj)                                            \
     if (!(cond)) {                                                   \
         bug_on((obj)->error, __FILE__, __LINE__, NULL);              \
         goto error;                                                  \
     }
-# define ensure0(cond, obj)                                          \
+#define ensure0(cond, obj)                                           \
     if (!(cond)) {                                                   \
```

Comment:
I don't think the use of `ensure0` is a big deal performance-wise. We can add some optimizations if that seems warranted, but for now, I'll commit this as is